src/app/missions/tasks/racing-task/racing-task.component.html

```diff
@@ -1,33 +1,81 @@
 <h4>{{task.uid}}: Racing Task (23)</h4>
 <p [ngSwitch]="task.taskParam1">
+  The player needs to
   <span *ngSwitchCase="1">
-    Be at or above the {{task.targetValue}}. place in the race world specified by {{task.target}}
+    finish in <b luxTooltip="targetValue">#{{task.targetValue}}</b> place or better in
+    <ng-container *ngTemplateOutlet="zones"></ng-container>.
   </span>
   <span *ngSwitchCase="2">
-    Achieve a {{task.targetValue}} ms lap time or better in the race world {{task.target}}
+    achieve a <b luxTooltip="targetValue">{{task.targetValue}}</b> ms lap time or better in
+    <ng-container *ngTemplateOutlet="zones"></ng-container>.
   </span>
   <span *ngSwitchCase="3">
-    Achieve a {{task.targetValue}} ms time or better in the race world  {{task.target}}
+    achieve a <b luxTooltip="targetValue">{{task.targetValue}}</b> ms total time or better in
+    <ng-container *ngTemplateOutlet="zones"></ng-container>.
+  </span>
+  <span *ngSwitchCase="4">
+    complete <b luxTooltip="targetValue">any {{task.targetValue}}</b> of the following missions:
+    <ng-container *ngFor="let target of task.targetGroup ? task.targetGroup.split(',') : []">
+      <a routerLink="/missions/{{target}}">{{target}}</a>,
+    </ng-container>
   </span>
   <span *ngSwitchCase="5">
-    Achieve {{task.targetValue}} achievements of the ones in {{task.targetGroup}}.
+    complete <b luxTooltip="targetValue">any {{task.targetValue}}</b> of the following missions:
+    <ng-container *ngFor="let target of task.targetGroup ? task.targetGroup.split(',') : []">
+      <a routerLink="/missions/{{target}}">{{target}}</a>,
```

Comment:
Can you replace all the trailing commas in ngFor things with a `<span class="inline-li">` container or sth that does `.inline-li + .inline-li::before { content: ", "; }`?

That way the comma isn't selectable but you don't need to special case the last element.

Reply:
Are we sure the comma shouldn't be selectable? Seems to me like making it unselectable would be surprising to the user.

Reply:
Yes, ideally it would just be normal, selectable text. You can use https://stackoverflow.com/a/52679602/4087068 to achieve that.

This was just a suggestion for an alternative way to get the same visual result, just with less markup and a little CSS.

Reply:
I ended up choosing the css solution, it's easier and allows us to change the display format if we want to in the future.